enterprise/server/cmd/ci_runner/main.go

```diff
@@ -1700,51 +1707,56 @@ func (ws *workspace) sync(ctx context.Context) error {
 	return nil
 }
 
-// fetchRefs fetches the pushed and target refs from their respective remotes
-func (ws *workspace) fetchRefs(ctx context.Context) error {
-	// "base" here is referring to the repo on which the workflow is configured.
-	baseRefs := make([]string, 0)
-	if *targetBranch != "" {
-		baseRefs = append(baseRefs, *targetBranch)
-	}
-	// "fork" is referring to the forked repo, if the runner was triggered by a
-	// PR from a fork (forkRefs will be empty otherwise).
-	forkRefs := make([]string, 0)
-
-	// Add the pushed ref to the appropriate list corresponding to the remote
-	// to be fetched (base or fork).
-	inFork := isPushedRefInFork()
-	if inFork {
-		// Try to fetch a branch if possible, because fetching
-		// commits that aren't at the tip of a branch requires
-		// users to set an additional config option (uploadpack.allowAnySHA1InWant)
-		pushedRefToFetch := *pushedBranch
-		if pushedRefToFetch == "" {
-			pushedRefToFetch = *commitSHA
-		}
-		forkRefs = append(forkRefs, pushedRefToFetch)
-	} else {
-		if *pushedBranch == "" {
-			baseRefs = append(baseRefs, *commitSHA)
-		} else if *pushedBranch != *targetBranch {
-			baseRefs = append(baseRefs, *pushedBranch)
-		}
+func (ws *workspace) shouldMergeBranches() bool {
```

Comment:
IMO this name is a bit misleading since it doesn't take `GetMergeWithBase()` into account. Not sure what a better name would be though - maybe "isMultiBranch()"? like, the workflow "involves" multiple branches (which may be fetched but not necessarily merged)